### ISSUE TITLE
recorded operations in the log may be deleted

### ISSUE THREAD

**Issue by hhy1999** [google/leveldb#1098]
Is it true that the log file in LevelDB records only the latest operations, and the recorded operations in the log may be deleted? I need to read the log file for backup purposes. Can I record all operations in the log file?

**Comment by cmumford** [comment]
You are correct. When the log gets to a certain size (default is [4MB](https://github.com/google/leveldb/blob/main/include/leveldb/options.h#L74-L82)) it is incorporated into the db and replaced with a new log file. https://github.com/google/leveldb/blob/main/doc/impl.md#log-files. I can't think of a way to record all operations unless it's done at the app level - i.e. create a wrapper and write your own log.

**Comment by hhy1999** [comment]
Thank you very much for answering my questions. I want to know how to parse the ldb file.There is an error in using the command：leveldbutil dump ，Corruption: corrupted compressed block contents.How can I solve this problem？




風逐梦
***@***.***



&nbsp;





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "google/leveldb"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月7日(星期二) 晚上8:16
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [google/leveldb] recorded operations in the log may be deleted (Issue #1098)





 
You are correct. When the log gets to a certain size (default is 4MB) it is incorporated into the db and replaced with a new log file. https://github.com/google/leveldb/blob/main/doc/impl.md#log-files. I can't think of a way to record all operations unless it's done at the app level - i.e. create a wrapper and write your own log.
